src/postject.cpp

```diff
@@ -0,0 +1,292 @@
+#include <algorithm>
+#include <codecvt>
+#include <locale>
+#include <memory>
+#include <vector>
+
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+
+#include <LIEF/LIEF.hpp>
+
+enum class ExecutableFormat { kELF, kMachO, kPE, kUnknown };
+
+enum class InjectResult { kAlreadyExists, kError, kSuccess };
+
+std::vector<uint8_t> vec_from_val(const emscripten::val& value) {
+  // NOTE - vecFromJSArray incurs a copy, so memory usage is higher than it
+  //        needs to be. Explore ways to access the memory directly and avoid
+  //        the copy.
+  return emscripten::vecFromJSArray<uint8_t>(value);
+}
```

Comment:
The docs say that `convertJSArrayToNumberVector` does no type checking and will silently change values, which could allow for subtle bugs (the script would run normally but the output could be slightly incorrect). While low risk, I'd prefer to have the type checking so an error occurs rather than a subtle behavior change for the moment. We can switch to `convertJSArrayToNumberVector` if we add our own type-checking (probably limit the input to be a `TypedArray`, for example) but for now I think this is fine. There's plenty of room for optimization (as the comment alludes to) in the Emscripten usage so let's roll that into a future PR optimizing stuff.